pkg/util/push/push.go

```diff
@@ -73,9 +74,14 @@ func handler(maxRecvMsgSize int,
 	allowSkipLabelNameValidation bool,
 	push Func,
 	parser parserFunc,
+	haRequestShortcuter func(context.Context, http.ResponseWriter, *http.Request) bool,
```

Comment:
Nit on the naming:
- we shouldn't say `ha` here. We don't care why the request should be shortcut. By saying `ha` we're making the contract stricter than we need.
- this is a function that returns a bool, which is used as `if ... {` so I'd say it's name should be `shouldShortcut` or more golang-idiomatic `shortcut`